integreat_cms/cms/views/pages/page_actions.py

```diff
@@ -827,3 +828,45 @@ def render_mirrored_page_field(request, region_slug, language_slug):
             "page_form": page_form,
         },
     )
+
+
+@permission_required("cms.change_page")
+# pylint: disable=unused-argument
+def refresh_date(
+    request,
+    page_id,
+    region_slug,
+    language_slug,
+):
+    """
+    Refresh the date for all translations of a corresponding page
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param page_id: The id of the page of the current page form
+    :type page_id: int
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :param language_slug: The slug of the current language
+    :type language_slug: str
+
+    :return: A redirection to the :class:`~integreat_cms.cms.views.pages.page_form_view.PageFormView`
+    :rtype: ~django.http.HttpResponseRedirect
+    """
+
+    page_translations = PageTranslation.objects.filter(page_id=page_id)
+    for page_translation in page_translations:
+        if page_translation is page_translation.latest_version:
+            page_translation.save()
```

Comment:
```suggestion
    page = get_object_or_404(request.region.pages, id=page_id)
    for page_translation in page.translations.distinct("page__pk", "language__pk"):
        page_translation.save()
```
The simple way to get all newest versions of all languages is a distinct query - this makes sure only one result is returned for each page/language combination, and since they are already ordered by version, the returned element will be the latest version.
Only remaining problem: Now the order of the translations are more or less random, but we need to save them in the correct order to preserve their state based on the region's language tree.
So the default language needs to be saved first, then the languages that are direct children of the default language and so on... I leave this task up to you :innocent: